src/index.ts

```diff
@@ -195,12 +195,16 @@ export class IFrameEthereumProvider extends EventEmitter<
    */
   private async execute<TParams, TResult, TErrorData>(
     method: string,
-    params?: TParams
+    params?: TParams,
+    optionalId?: number
   ): Promise<
     | JsonRpcSucessfulResponseMessage<TResult>
     | JsonRpcErrorResponseMessage<TErrorData>
   > {
-    const id = getUniqueId();
+    const id = Number.isNaN(optionalId as any)
```

Comment:
Here we should check for undefined values, if `optionalId` value is `undefined` then `id` will be `undefined` too